biodatasets/tmvar/tmvar.py

```diff
@@ -0,0 +1,327 @@
+                                    datasets.Value("string")
+                                ),
+                            }
+                        ],
+                    }
+                )
+            elif version == "v2":
+                features = datasets.Features(
+                    {
+                        "pmid": datasets.Value("string"),
+                        "passages": [
+                            {
+                                "type": datasets.Value("string"),
+                                "text": datasets.Sequence(datasets.Value("string")),
+                                "offsets": datasets.Sequence([datasets.Value("int32")]),
+                            }
+                        ],
+                        "entities": [
+                            {
+                                "text": datasets.Sequence(datasets.Value("string")),
+                                "offsets": datasets.Sequence([datasets.Value("int32")]),
+                                "concept_id": datasets.Value("string"),
+                                "semantic_type_id": datasets.Sequence(
+                                    datasets.Value("string")
+                                ),
+                                "rsid": datasets.Value("string"),
+                            }
+                        ],
+                    }
+                )
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        version = self.config.name.split("_")[1]
+        # Hacky fix for inconsitent data format in tmvar
+        # v1 is a zip of train and test files
+        # v2 is a single .txt file with only the test data
+        urls = _URLS[_DATASETNAME][version]
+        if version == "v1":
+            data_dir = dl_manager.download_and_extract(urls)
+            train_filename = "tmVarCorpus\\train.PubTator.txt"
+            test_filename = "tmVarCorpus\\test.PubTator.txt"
+            train_filepath = os.path.join(data_dir, train_filename)
+            test_filepath = os.path.join(data_dir, test_filename)
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={
+                        "filepath": train_filepath,
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={
+                        "filepath": test_filepath,
+                    },
+                ),
+            ]
+        elif version == "v2":
+            data_dir = dl_manager.download(urls)
```

Comment:
when you use `download` instead of `download_and_extract` the thing that is returned is the path to the file not a directory ... try `test_filepath = dl_manager.download(urls)`

full docs here https://github.com/huggingface/datasets/blob/master/src/datasets/utils/download_manager.py#L259

Reply:
Thank you! That worked

Reply:
(I think the above still works) but I don't think it's the difference between download and extract .. I think its just the fact that one string went in so one string comes out ... if you passed in `[urls]` then you would get a single element list back with the path to the downloaded file.